client/components/chat.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Card from 'react-bootstrap/Card';
+
+export default class Chat extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      content: '',
+      userId: null,
+      roomId: null
+    };
+    this.handleMessageInput = this.handleMessageInput.bind(this);
+    this.messageSend = this.messageSend.bind(this);
+  }
+
+  handleMessageInput(event) {
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    this.setState({
+      [name]: value,
+      userId: this.props.userId,
+      roomId: this.props.roomId
+    });
+  }
+
+  messageSend(event) {
+    event.preventDefault();
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
```

Comment:
Here you can grab what you need from state and props. It's best to avoid putting extra stuff in state that you could just grab from props on-demand.